tests/src/astro/electromagnetism/unitTestYarkovskyAcceleration.cpp

```diff
@@ -0,0 +1,207 @@
+            { "Sun", { pointMassGravityAcceleration( ), yarkovskyAcceleration( yarkovskyParameter ) }},
+    };
+
+    const double sunGravitationalParameter = bodies.at( "Sun" )->getGravitationalParameter( );
+
+    // Make the acceleration models
+    SelectedAccelerationMap accelerationSettings {{ "Asteroid", accelerationsOfAsteroid }};
+    basic_astrodynamics::AccelerationMap accelerationModels = createAccelerationModelsMap( bodies,
+                                                                                           accelerationSettings,
+                                                                                           bodiesToPropagate,
+                                                                                           centralBodies );
+
+    // Initial State
+    Eigen::VectorXd initialState = orbital_element_conversions::convertKeplerianToCartesianElements(
+            initialSemiMajorAxis,
+            initialEccentricity,
+            initialInclination,
+            initialRAAN,
+            initialArgOfPeri,
+            initialTrueAnomaly,
+            sunGravitationalParameter );
+
+    // Termination Setting
+    std::shared_ptr< propagators::PropagationTerminationSettings > terminationSettings = propagators::propagationTimeTerminationSettings(
+            simulationEndEpoch,
+            true );
+
+    // Integrator Settings
+    std::shared_ptr< numerical_integrators::IntegratorSettings< double>> integratorSettings = numerical_integrators::rungeKutta4Settings(
+            fixedStepSize );
+
+    // Propagator Settings
+    std::shared_ptr< propagators::TranslationalStatePropagatorSettings< double, double>> propagatorSettings = propagators::translationalStatePropagatorSettings(
+            centralBodies,
+            accelerationModels,
+            bodiesToPropagate,
+            initialState,
+            simulationStartEpoch,
+            integratorSettings,
+            terminationSettings,
+            propagators::cowell );
+
+    // Dynamics Simulator
+    propagators::SingleArcDynamicsSimulator< > dynamicsSimulator( bodies, propagatorSettings );
+
+    // State history
+    std::map< double, Eigen::VectorXd > stateHist = dynamicsSimulator.getEquationsOfMotionNumericalSolution( );
+
+    // Calculate Expected Drift
+    Eigen::Vector6d keplerFinalState = orbital_element_conversions::convertCartesianToKeplerianElements( Eigen::Vector6d(
+            stateHist.at( simulationEndEpoch )), sunGravitationalParameter );
+
+    const double& a = initialSemiMajorAxis;
+    const double& e = initialEccentricity;
+
+    const double p = a * ( 1 - e * e ); // Semi-latus rectum
+    const double n = std::sqrt( sunGravitationalParameter / ( a * a * a )); // Mean motion
+
+    const double expectedSemiMajorAxisDrift =
+            ( 2 * yarkovskyParameter * ( 1 - e * e ) * AU * AU / ( n * p * p )) * simulationDuration;
+    const double calculatedSemiMajorAxisDrift = keplerFinalState[0] - initialSemiMajorAxis;
```

Comment:
In principle, this check is reasonable, but it may cause issues since the time-history of semi-major axis may include small (once-per-orbit) variability. Depending on where in the phase of this variability you are, your result here may be biased. A more robust approach would be to take the time history of semi-major axis (save as dependent variable) and do linear fit of the values. A very similar thing is done here:

https://github.com/tudat-team/tudat/blob/master/tests/src/astro/gravitation/unitTestDirectTidalDissipationAcceleration.cpp#L139

One difference: the manner in which the semiMajorAxes map in the above example is obtained is very convoluted. You can just save the kepler elements as dependent variables, iterate over the dependent variables at each epoch, and create the semi-major axis map in that way.

Reply:
I thought an alternative solution would be to propagate exactly one orbit. The variability then averages out. Does this make sense?